BREAKING CHANGE: add eslint9 config files (#125)
Brightspace/eslint-config-brightspace: README.md

## Usage

Shared configurations can be directly exported or included as part of a custom configuration in the `eslint.config..js` file.
```js
export { nodeConfig as default } from './index.js';
```

```js
import { nodeConfig } from 'eslint-config-brightspace';

export default [
	...nodeConfig,
	// Custom configuration
];
```

> 

Reviewer: Should these two use cases maybe be split out to make it more clear what the options are? Also shouldn't the export be from `eslint-config-brightspace`? Something like:

Shared configurations can be directly exported from `eslint.config.js`:
```js
export { nodeConfig as default } from 'eslint-config-brightspace';
```

Or included as part of a custom configuration in `eslint.config..js`:
```js
import { nodeConfig } from 'eslint-config-brightspace';

export default [
	...nodeConfig,
	// Custom configuration
];
```

> ### Utils
> 
> Since the `--ext` flag is deprecated, include extensions by using the `addExtenstion` function from `utils.js`.
> ```js
> import { nodeConfig } from 'eslint-config-brightspace';
> 
> export default addExtensions(nodeConfig, ['.js','.html']);
> ```

Reviewer: The fact that `--ext` is deprecated will become irrelevant and forgotten pretty soon. I'd suggest just describing each utility's purpose. Something like:

```suggestion
### Additional File Extensions

Include extensions beyond `.js` files using the `addExtenstion` function from `utils.js`:
```js
import { addExtensions } from 'eslint-config-brightspace/utils.js';
import { nodeConfig } from 'eslint-config-brightspace';

export default addExtensions(nodeConfig, ['.js','.html']);
```
```

```

```json
{
  "extends": "brightspace/<environment>"

> }

Reviewer: Maybe prefix this section with a "Different Configurations for Different Directories" heading?

export default setDirectoryConfigs(
	litConfig,
	{
		'test': testingConfig,
		'test/cli': nodeConfig
	}
);
```
Note that each set configuration will force all prior configurations to ignore it. For example, for the above configuration, `litConfig` will ignore any files in the `test` directory; and `testingConfig` will ignore any files in the `test/cli` directory.

### Environment Specific Configs

| Environment | Description |
|--|--|
| `browserConfig` | use with code that runs in a browser |
| `litConfig` | use with [Lit](https://lit.dev/) projects |
| `testingConfig` | use with [@brightspace-ui/testing](https://github.com/BrightspaceUI/testing) test code |
| `nodeConfig` | use with [Node.js](https://nodejs.org) projects |
| `reactConfig` | use with [React](https://react.dev/) projects |

Example:

```js

> export { nodeConfig as default } from './index.js';

Reviewer: ```suggestion
export { nodeConfig as default } from 'eslint-config-brightspace';
```

## Usage

> Shared configuration, they need to exported from the `eslint.config.js` file on the root directory.

Reviewer: ```suggestion
To use a shared configuration without any customizations, export it from a `eslint.config.js` file at the project root:
```

```js

> export { nodeConfig as default } from 'eslint-config-brightspace/index.js';

Reviewer: `index.js` shouldn't be necessary since it's the main entrypoint for the package.
```suggestion
export { nodeConfig as default } from 'eslint-config-brightspace';
```

```

> They can also be included as part of a custom configuration.

Reviewer: ```suggestion
Shared configurations can also be included in a custom configuration:
```

```js
import { nodeConfig } from 'eslint-config-brightspace';

export default [
	...nodeConfig,
	// Custom configuration
];
```

### Additional File Extensions

Include extensions beyond `.js` files using the `addExtenstion` helper function:
```js
import { addExtensions, nodeConfig } from 'eslint-config-brightspace';
export default addExtensions(nodeConfig, ['.js','.html']);
```

### Different Configurations for Different Directories

> To include different configurations for specific directories, use the `setDirectoryConfigs` helper function. This replaces the [configuration hierarchy](https://eslint.org/docs/v8.x/use/configure/configuration-files#cascading-and-hierarchy) from `eslint8`. To use it, include the global configuration and specify the directory configurations, these will apply to all files inside the directory and recursively to any of its subdirectories.

Reviewer: ```suggestion
To include different configurations for specific directories, use the `setDirectoryConfigs` helper function. This replaces the [configuration hierarchy](https://eslint.org/docs/v8.x/use/configure/configuration-files#cascading-and-hierarchy) from `eslint8`.

Include the global configuration and specify the directory configurations. These will apply to all files inside the directory and recursively to any of its subdirectories.
```

```js
import { litConfig, nodeConfig, setDirectoryConfigs, testingConfig } from 'eslint-config-brightspace';

export default setDirectoryConfigs(
	litConfig,
	{
		'test': testingConfig,
		'test/cli': nodeConfig
	}
);
```
Note that each set configuration will force all prior configurations to ignore it. For example, for the above configuration, `litConfig` will ignore any files in the `test` directory; and `testingConfig` will ignore any files in the `test/cli` directory.

### Environment Specific Configs

| Environment | Description |
|--|--|
| `browserConfig` | use with code that runs in a browser |
| `litConfig` | use with [Lit](https://lit.dev/) projects |
| `testingConfig` | use with [@brightspace-ui/testing](https://github.com/BrightspaceUI/testing) test code |
| `nodeConfig` | use with [Node.js](https://nodejs.org) projects |
| `reactConfig` | use with [React](https://react.dev/) projects |

Example:

```js

> export { nodeConfig as default } from 'eslint-config-brightspace/index.js';

Reviewer: ```suggestion
export { nodeConfig as default } from 'eslint-config-brightspace';
```
